solidity/test/token/HypERC20CollateralVaultDeposit.t.sol

```diff
@@ -0,0 +1,244 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.13;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+import "forge-std/Test.sol";
+import {ERC4626Test} from "../../contracts/test/ERC4626/ERC4626Test.sol";
+import {TypeCasts} from "../../contracts/libs/TypeCasts.sol";
+import {HypTokenTest} from "./HypERC20.t.sol";
+
+import {HypERC20CollateralVaultDeposit} from "../../contracts/token/HypERC20CollateralVaultDeposit.sol";
+import "../../contracts/test/ERC4626/ERC4626Test.sol";
+
+contract HypERC20CollateralVaultDepositTest is HypTokenTest {
+    using TypeCasts for address;
+    uint constant DUST_AMOUNT = 1e11;
+    HypERC20CollateralVaultDeposit internal erc20CollateralVaultDeposit;
+    ERC4626Test vault;
+
+    function setUp() public override {
+        super.setUp();
+        vault = new ERC4626Test(address(primaryToken), "Regular Vault", "RV");
+
+        localToken = new HypERC20CollateralVaultDeposit(
+            vault,
+            address(localMailbox)
+        );
+        erc20CollateralVaultDeposit = HypERC20CollateralVaultDeposit(
+            address(localToken)
+        );
+
+        erc20CollateralVaultDeposit.enrollRemoteRouter(
+            DESTINATION,
+            address(remoteToken).addressToBytes32()
+        );
+
+        remoteMailbox.setDefaultHook(address(noopHook));
+        remoteMailbox.setRequiredHook(address(noopHook));
+        primaryToken.transfer(ALICE, 1000e18);
+        _enrollRemoteTokenRouter();
+    }
+
+    function testERC4626VaultDeposit_RemoteTransfer_deposits_intoVault(
+        uint256 transferAmount
+    ) public {
+        vm.assume(transferAmount < TOTAL_SUPPLY);
+
+        vm.startPrank(ALICE);
+        primaryToken.mint(transferAmount);
+        primaryToken.approve(address(localToken), transferAmount);
+        vm.stopPrank();
+
+        // Check vault shares balance before and after transfer
+        assertEq(vault.maxRedeem(address(erc20CollateralVaultDeposit)), 0);
+        assertEq(erc20CollateralVaultDeposit.assetDeposited(), 0);
+
+        _performRemoteTransfer(0, transferAmount);
+        assertApproxEqAbs(
+            vault.maxRedeem(address(erc20CollateralVaultDeposit)),
+            transferAmount,
+            1
+        );
+        assertEq(erc20CollateralVaultDeposit.assetDeposited(), transferAmount);
+    }
+
+    function testERC4626VaultDeposit_RemoteTransfer_withdraws_fromVault(
+        uint256 transferAmount
+    ) public {
+        vm.assume(transferAmount < TOTAL_SUPPLY);
+
+        // Transfer to Bob
+        vm.startPrank(ALICE);
+        primaryToken.mint(transferAmount);
+        primaryToken.approve(address(localToken), transferAmount);
+        vm.stopPrank();
```

Comment:
make a helper for this? or move to setup?

Reply:
Moved this and other duplicate calls to their own helpers